app/org/sagebionetworks/bridge/stormpath/StormpathAccountDao.java

```diff
@@ -130,63 +119,16 @@ public Account verifyEmail(StudyIdentifier study, EmailVerification verification
         checkNotNull(email);
-        
-        // This is painful, it's not in the Java SDK. I hope we can come back to this when it's in their SDK
-        // and move it over.
-        SimpleHttpConnectionManager manager = new SimpleHttpConnectionManager();
-        int status = 202; // The Stormpath resend method returns 202 "Accepted" when successful
-        byte[] responseBody = new byte[0];
-        try {
-            BridgeConfig config = BridgeConfigFactory.getConfig();
-            
-            String bodyJson = "{\"login\":\""+email.getEmail()+"\"}";
-            
-            HttpClient client = new HttpClient(manager);
-            
-            PostMethod post = new PostMethod(this.application.getHref() + "/verificationEmails");
-            post.setRequestHeader("Accept", "application/json");
-            post.setRequestHeader("Content-Type", "application/json");
-            post.setRequestHeader("Bridge-Study", studyIdentifier.getIdentifier());
-            post.setRequestEntity(new StringRequestEntity(bodyJson, "application/json", "UTF-8"));
-
-            UsernamePasswordCredentials creds = new UsernamePasswordCredentials(
-                    config.getStormpathId().trim(), config.getStormpathSecret().trim());
-            
-            client.getState().setCredentials(ANY, creds);
-            client.getParams().setParameter(AuthPolicy.AUTH_SCHEME_PRIORITY, Lists.newArrayList(AuthPolicy.DIGEST));
-            client.getParams().setAuthenticationPreemptive(true);
-            
-            status = client.executeMethod(post);
-            responseBody = post.getResponseBody();
-
-        } catch(ResourceException e) {
-            rethrowResourceException(e, null);
-        } catch(Throwable throwable) {
-            throw new BridgeServiceException(throwable);
-        } finally {
-            manager.shutdown();
-        }
-        // If it *wasn't* a 202, then there should be a JSON message included with the response...
-        if (status != 202) {
-
-            // One common response, that the email no longer exists, we have mapped to a 404, so do that 
-            // here as well. Otherwise we treat it on the API side as a 503 error, a service unavailable problem.
-            JsonNode node = null;
-            try {
-                node = BridgeObjectMapper.get().readTree(responseBody);    
-            } catch(IOException e) {
-                throw new BridgeServiceException(e);
-            }
-            String message = node.get("message").asText();
-            if (message.contains("does not match a known resource")) {
-                throw new EntityNotFoundException(Account.class);
-            }
-            throw new ServiceUnavailableException(message);
-        }
+        VerificationEmailRequestBuilder requestBuilder = Applications.verificationEmailBuilder();
+        VerificationEmailRequest request = requestBuilder
+                .setAccountStore(application.getDefaultAccountStore())
```

Comment:
It needs to use the directory (account store) for the study.... we can get this, the study has the HREF of that directory.